### DISCUSSION TITLE
Linker error using ImPlot with ImGui v1.88 in Unreal Engine 5

### DISCUSSION THREAD

**Discussion by mmannj** [ocornut/imgui#5489] [New Users Build/Link/Run issues ONLY!]
I am using ImGui with the newest version of ImPlot in Unreal Engine 5. And the project compiles and builds with ImGui version 1.87 but I am getting a linker error when I update to ImGui v1.88.
An linker error occurs whenever I reference something from ImPlot, and I am not getting any compiler errors.

My build file is unchanged, has anyone encountered this issue?

 error LNK2019: unresolved external symbol "__declspec(dllimport) bool __cdecl ImPlot::BeginPlot(char const *,struct ImVec2 const &,int)"

if I comment out "PrivateDefinitions.Add(string.Format("IMPLOT_API=DLLIMPORT"));" from my project' build file, the linker error is:
 error LNK2019: unresolved external symbol "bool __cdecl ImPlot::BeginPlot(char const *,struct ImVec2 const &,int)"

I do not know if I need a micro in ImPlot for the newer version of ImGui.

**Comment by mmannj** [comment]
I had to include the ImPlot implementation files to ImGuiImplementatio.cpp

And create ImPlot context in ImGuiContextProxy.cpp

**Reply by ninonick0607** [reply]
Hey I know this was a message from a while ago but do you happen to know how to integrate implot with unreal? Ive got imgui working but implot has been a struggle. Thanks.

**Reply by ocornut** [reply]
The Unreal backends listed here seem to incorporate ImPlot:https://github.com/ocornut/imgui/wiki/Bindings#frameworkengine-backends
